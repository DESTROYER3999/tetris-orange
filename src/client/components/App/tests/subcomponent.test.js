import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import AppSub from "../subcomponent";
import socket from "../../../services/socket-api";

beforeEach(done => {
  // Setup
  if (socket.disconnected) {
    socket.connect();
  }
  socket.on("connect", () => {
    done();
  });
});

afterEach(done => {
  //   Cleanup;
  if (socket.connected) {
    socket.disconnect();
  }
  done();
});

describe("<LoginSub />", () => {
  test("App rendering with a player in game", () => {
    const wrapper = shallow(<AppSub playerName="Adubedat" roomName="Room1" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("App rendering with a player in Lobby", () => {
    const wrapper = shallow(<AppSub playerName="Adubedat" roomName="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test("App rendering with an unconnected user", () => {
    const wrapper = shallow(<AppSub playerName="" roomName="" />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
