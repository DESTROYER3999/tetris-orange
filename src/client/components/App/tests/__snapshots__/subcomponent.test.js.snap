// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginSub /> App rendering with a player in Lobby 1`] = `
<styled.div
  id="App"
>
  <Connect(HeaderSub) />
  <Connect(LobbySub) />
</styled.div>
`;

exports[`<LoginSub /> App rendering with a player in game 1`] = `
<styled.div
  id="App"
>
  <Connect(HeaderSub) />
  <Connect(GameSub) />
</styled.div>
`;

exports[`<LoginSub /> App rendering with an unconnected user 1`] = `
<styled.div
  id="App"
>
  <Connect(HeaderSub) />
  <Connect(LoginSub) />
</styled.div>
`;
